backend/assignments/1/src/main/java/org/example/Main.java

```diff
@@ -0,0 +1,290 @@
+package org.example;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Scanner;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Main class for the Crypto Trading Application.
+ */
+public class Main {
+
+    private static final Logger logger = Logging.logger;
+
+    private static List<Coin> coins;
+    private static List<Trader> traders;
+
+    /**
+     * Main method to initiate the Crypto Trading Application.
+     *
+     * @param args Command-line arguments
+     */
+    public static void main(String[] args) throws InterruptedException {
+        // Step 1: Loading data from CSV files into memory
+        loadCsvData();
+
+        // Step 2: Loading JSON transaction data into memory
+        JsonNode jsonTransactions = loadJsonTransactions("test_transaction.json");
+
+        // Step 3: Execute transactions using a thread pool
+        assert jsonTransactions != null;
+        executeTransactions(jsonTransactions);
+
+        // Step 4: Menu-Driven Frontend
+        displayMenu();
+    }
+
+    /**
+     * Loads coin and trader data from CSV files into memory.
+     */
+    private static void loadCsvData() {
+        String coinsFilePath = "coins.csv";
+        String tradersFilePath = "traders.csv";
+
+        try {
+            coins = CsvLoader.loadCoins(coinsFilePath);
+            traders = CsvLoader.loadTraders(tradersFilePath);
+
+            if (coins == null || traders == null) {
```

Comment:
`if(!coin)` can suffice.
Also, using `Objects.isNull()` for null checks is cleaner